src/ReorderEligibility/ItemsOutOfStockEligibilityChecker.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\CustomerReorderPlugin\ReorderEligibility;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+
+final class ItemsOutOfStockEligibilityChecker implements ReorderEligibilityChecker
+{
+    /** @var ReorderEligibilityConstraintMessageFormatterInterface */
+    private $reorderEligibilityConstraintMessageFormatter;
+
+    public function __construct(
+        ReorderEligibilityConstraintMessageFormatterInterface $reorderEligibilityConstraintMessageFormatter
+    ) {
+        $this->reorderEligibilityConstraintMessageFormatter = $reorderEligibilityConstraintMessageFormatter;
+    }
+
+    public function check(OrderInterface $order, OrderInterface $reorder): array
+    {
+        $variantsOutOfStock = [];
+
+        /** @var OrderItemInterface $orderItem */
+        foreach ($order->getItems()->getValues() as $orderItem) {
+            /** @var ProductVariantInterface $productVariant */
+            $productVariant = $orderItem->getVariant();
+            if ($productVariant->isTracked() && !$productVariant->isInStock()) {
+                array_push($variantsOutOfStock, $orderItem->getVariantName());
```

Comment:
The preferred way to add something to an array is `$variantsOutOfStock[] = $orderItem->getVariantName()`